cs-multi-shell/src/main.ts

```diff
@@ -4,10 +4,21 @@ import { IonicRouteStrategy, provideIonicAngular } from '@ionic/angular/standalo
 
 import { routes } from './app/app.routes';
 import { AppComponent } from './app/app.component';
+import { AuthGuardService } from './app/auth-guard.service';
+import { APP_INITIALIZER } from '@angular/core';
+import { VaultService } from './app/vault.service';
+
+const appInitFactory =
+  (vaultService: VaultService): (() => Promise<void>) =>
+    async () => {
+      await vaultService.init();
+    }
 
 bootstrapApplication(AppComponent, {
   providers: [
     { provide: RouteReuseStrategy, useClass: IonicRouteStrategy },
+    { provide: APP_INITIALIZER, useFactory: appInitFactory, deps: [VaultService], multi: true },
```

Comment:
Important: We use the APP_INITIALIZER in Angular to ensure our vault is initialized before doing anything else. If we didnt do this we would have to ensure that every constructor and route did not access the vault before it was intialized.